repo/fsrepo/fsrepo.go

```diff
@@ -595,11 +596,43 @@ func (r *FSRepo) GetConfigKey(key string) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	var cfg map[string]interface{}
-	if err := serialize.ReadConfigFile(filename, &cfg); err != nil {
+
+	var cfgMap map[string]interface{}
+	if err := serialize.ReadConfigFile(filename, &cfgMap); err != nil {
+		return nil, err
+	}
+
+	/*
+		Get an instance of Config for retrieving Config defined keys. Grabbing
+		values directly from Config allows us to retrieve values for keys that
+		may be flagged as 'json:",omitempty"' when the values of those keys are
+		considered empty.
+	*/
+	cfg, err := config.FromMap(cfgMap)
+	if err != nil {
 		return nil, err
 	}
-	return common.MapGetKV(cfg, key)
+
+	// Walk the Config instance looking for key fields
+	fieldValue := reflect.ValueOf(cfg)
+	parts := strings.Split(key, ".")
+	for i, part := range parts {
+		fieldValue = reflect.Indirect(fieldValue).FieldByName(part)
+
+		// Break out if the current field is not a defined key in Config
+		if !fieldValue.IsValid() {
```

Comment:
I'm not sure what is being tested here. When would a decoded field here not be valid? AFAIK (which is not that much) you can only get an invalid `reflect.Value` from _within_ reflect itself by some weird manipulation. See https://stackoverflow.com/a/39011500 for nice examples.

If this is trying to check if there's an empty/nil value here because we didn't pick up anything from the JSON, then you might want to go with something like https://stackoverflow.com/a/23555352.

Reply:
Solution direction changed, resolving conservation.